lib/options.go

```diff
@@ -521,6 +598,20 @@ func (o Options) Apply(opts Options) Options {
 		o.ConsoleOutput = opts.ConsoleOutput
 	}
 
+	// FIXME: This should really be in applyDefault(), but in some tests
+	// o.DNS hasn't been initialized yet...
+	if o.DNS == nil {
+		o.DNS = &DefaultDNSConfig
+	}
```

Comment:
so ... here we get the address of the default DNS config and then ... change it(the config, not the address)? This doesn't seem like a good idea and will likely break with tests ... especially parallel ones

Reply:
Not an issue anymore since I switched to using a struct value for `Options.DNSConfig` and not a pointer.